### PR TITLE
Misc Annoy index improvements

### PR DESCRIPTION
**Pull request by rschu1ze**
A mixed bag of fixes for Annoy:
- tests:
  -  added tests with different `GRANULARITY`-s.
  - re-enabled tests which were disabled (#50714) due to flakiness in analyzer tests 
  - commented out tests with unstable results. This was caused by the inherent randomness of Annoy search. They are useful anyways, so don't remove them completely.
- much more extensive docs
- changed the default `GRANULARITY` of Annoy indexes from 1 to 100 million rows. This is much closer to the behavior users intuitively expect

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 0e2fd8744ed8d849bb763ea46991c205056b1bd1 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/50661/0e2fd8744ed8d849bb763ea46991c205056b1bd1/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by rschu1ze** [comment]
ClickHouse Stateless Tests (debug) [4/5]:
- 01605_adaptive_granularity_block_borders #50681

ClickHouse Stateless Tests (msan) [3/6]
- 00534_functions_bad_arguments12 #50619
- 00534_functions_bad_arguments4_long #50619

File: tests/queries/0_stateless/02354_annoy_index.sql [hunk 1]
```diff
@@ -1,150 +1,251 @@
...
+SELECT 'ORDER BY type, L2Distance, check that index is used';
+EXPLAIN indexes=1
+SELECT *
+FROM tab
+ORDER BY L2Distance(vector, [10.0, 0.0, 10.0, 0.0])
+LIMIT 3;
+
+DROP TABLE tab;
+
+-- (*) Storage and search in Annoy indexes is inherently random. Tests which check for exact row matches would be unstable. Therefore,
```
Comment (devcrafter) [inline tests/queries/0_stateless/02354_annoy_index.sql:250]: AFAIS, Annoy implementation supports `void set_seed(R q)` method. Probably, we could use to generate repeatable index. So, logically, using it and disabling merges can allow getting stable results for the queries

**Review by devcrafter — APPROVED** [review]
@rschu1ze Please check the comments regarding the tests. It's up to you when address them, if at all

**Comment by rschu1ze** [comment]
@devcrafter Very good comments, thanks. I'd look into them separately. For now, allow me to merge - someone is waiting for the updated docs and I don't want to point to an open PR.

**Comment by rfraposa** [comment]
Just FYI: this PR has broken the build of the docs. I am going to try and fix the issue in a separate PR.

**Comment by devcrafter** [comment]
> @devcrafter Very good comments, thanks. I'd look into them separately. For now, allow me to merge - someone is waiting for the updated docs and I don't want to point to an open PR.

@rschu1ze Please add me as a reviewer to follow-up PRs

**Comment by rschu1ze** [comment]
> Just FYI: this PR has broken the build of the docs. I am going to try and fix the issue in a separate PR.

Sorry for the hassle (I was pretty sure that the docs check was green).

> @rschu1ze Please add me as a reviewer to follow-up PRs

Will do, here is one: #50912 ... more are coming.

**Comment by rfraposa** [comment]
No hassle at all...happy to fix it. Docusaurus uses React and it gets confused when it sees something like `<ann_index_name>`. I just replaced the angle brackets with square brackets

**Pull request merged**